src/tests/test_compose.py

```diff
@@ -7,12 +11,129 @@ def test_compose(ip):
 
     ip.run_cell_magic(
         "sql",
-        "--with author_sub --save final",
+        "--save final",
         "SELECT last_name FROM author_sub;",
     )
-
+    out, _ = capsys.readouterr()
     result = ip.run_cell("%sqlrender final").result
-    expected = "WITH `author_sub` AS (\nSELECT last_name \
-FROM author WHERE year_of_death > 1900)\nSELECT last_name FROM author_sub;"
+    expected = (
+        "WITH `author_sub` AS (\nSELECT last_name FROM author "
+        "WHERE year_of_death > 1900)\nSELECT last_name FROM author_sub;"
+    )
 
     assert result == expected
+    assert "Forming CTE from saved snippets : author_sub" in out
+
+
+def test_deprecation_warning(ip):
+    ip.run_cell_magic(
+        "sql",
+        "--save author_sub",
+        "SELECT last_name FROM author WHERE year_of_death > 1900",
+    )
+
+    with pytest.warns(FutureWarning) as record:
+        ip.run_cell_magic(
+            "sql",
+            "--with author_sub --save final",
+            "SELECT last_name FROM author_sub;",
+        )
+        assert len(record) == 1
+        assert (
+            "--with will be deprecated soon, and dependencies "
+            "will be inferred from saved queries instead" in record[0].message.args[0]
+        )
```

Comment:
the assert statements should outside of the `with` context manager

Reply:
removed since it is print statements now instead of warning